### PR TITLE
feat(buttons): Add icon button variants.

### PR DESCRIPTION
**Pull request by GerardPaligot**
## Changes description

Add icon button variants

## Context

A new variant has been added for buttons with an icon only

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.
- [x] I have tested on a tablet device/emulator.
- [x] I have tested on a large screen device/emulator.
- [ ] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

#### Phone

image

### CODE REVIEW

File: buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+private val ICON_BUTTON_LARGE_ATTR = R.attr.vtmnIconButtonLargeSize
+
+open class VitaminPrimaryMediumIconButton @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.vtmnIconButtonPrimaryMediumStyle
+) : MaterialButton(context, attrs, defStyleAttr) {
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        layoutParams.width = context.getDimensionPixelSizeFromAttr(ICON_BUTTON_MEDIUM_ATTR)
```
Comment (MaximeJineMaraval) [inline buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt:26]: Why is this needed? The XML declaration is not working as usual?

File: buttons/src/main/res/values/attrs.xml [hunk 2]
```diff
@@ -15,6 +15,24 @@
...
+    <attr name="vtmnIconButtonConversionMediumStyle" format="reference"/>
+    <attr name="vtmnIconButtonConversionLargeStyle" format="reference"/>
+    <attr name="vtmnIconButtonPrimaryReversedMediumStyle" format="reference"/>
+    <attr name="vtmnIconButtonPrimaryReversedLargeStyle" format="reference"/>
+    <attr name="vtmnIconButtonGhostMediumStyle" format="reference"/>
+    <attr name="vtmnIconButtonGhostLargeStyle" format="reference"/>
+    <attr name="vtmnIconButtonGhostReversedMediumStyle" format="reference"/>
+    <attr name="vtmnIconButtonGhostReversedLargeStyle" format="reference"/>
+
+    <attr name="vtmnIconButtonMediumSize" format="dimension" />
```
Comment (MaximeJineMaraval) [inline buttons/src/main/res/values/attrs.xml:33]: Why declare this as an attr? Is it not working with just a simple dimens?

File: buttons/src/main/res/values/styles-icon-buttons.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="Widget.Vitamin.IconButton.Primary.Medium" parent="@style/Widget.MaterialComponents.Button.UnelevatedButton">
+        <item name="android:layout_width">@dimen/vtmn_icon_buttons_size_medium</item>
```
Comment (MaximeJineMaraval) [inline buttons/src/main/res/values/styles-icon-buttons.xml:4]: An attribute is declared for the kotlin code but the dimension is used here.
Is it possible to have the same code everywhere? (only dimens or only attribute if it's mandatory)

**Review by MaximeJineMaraval — CHANGES_REQUESTED** [review]

**Comment by MaximeJineMaraval** [comment]
> 🛑 Not really sure about the `right icon` variant, It looks non-aligned. Can we check it?

The `right icon` has not been modified in this contribution. If there is a real bug with it, it should be in another issue / pull request 😃

File: buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt (hunk 1 above)
Comment (GerardPaligot) [reply buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt:26]: No. If you don't force the size here, the button add padding of the text around the icon.

Capture d’écran 2022-06-23 à 15 07 25

File: buttons/src/main/res/values/styles-icon-buttons.xml (hunk 3 above)
Comment (GerardPaligot) [reply buttons/src/main/res/values/styles-icon-buttons.xml:4]: I can use the reference or the dimension. Let's take a decision about the reference and I'll update this line according to the final result.

File: buttons/src/main/res/values/attrs.xml (hunk 2 above)
Comment (MaximeJineMaraval) [reply buttons/src/main/res/values/attrs.xml:33]: You can customize it without attr. Simply write a dimens with the same name in your project will override it

File: buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt (hunk 1 above)
Comment (MaximeJineMaraval) [reply buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt:26]: So these lines in the ReadMe are wrong, isn't it? 

```xml
<com.google.android.material.button.MaterialButton
    style="?attr/vtmnIconButtonPrimaryMediumStyle"
    android:layout_width="?attr/vtmnIconButtonMediumSize"
    android:layout_height="?attr/vtmnIconButtonMediumSize"
    app:icon="@drawable/ic_vtmn_add" />
```

It will render the button the same as in your screenshot above?

File: buttons/src/main/res/values/attrs.xml (hunk 2 above)
Comment (GerardPaligot) [reply buttons/src/main/res/values/attrs.xml:33]: Yes but you need to know the name of this dimen which is internal. I don't think it is the best way.

File: buttons/src/main/res/values/styles-icon-buttons.xml (hunk 3 above)
Comment (GerardPaligot) [reply buttons/src/main/res/values/styles-icon-buttons.xml:4]: Totally agree

File: buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt (hunk 1 above)
Comment (GerardPaligot) [reply buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt:26]: Not exactly because I give the `layout_width` and the `layout_height` too.

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/62ac2962b2dd627a/build/1b383b48-08da-45be-905d-4934da2928cd/artifact/030ef7b03192f748/p/61b9b8e223f0a6d2ef8bf4324ad053c0

**Review by MaximeJineMaraval — APPROVED** [review]
